src/main/java/org/dependencytrack/model/Vulnerability.java

```diff
@@ -284,6 +284,24 @@ public static Source resolve(String id) {
     @Pattern(regexp = RegexSequence.Definition.PRINTABLE_CHARS_PLUS, message = "The CVSSv3 Vector may only contain printable characters")
     private String cvssV3Vector;
 
+    @Persistent
+    @Column(name = "CVSSV4BASESCORE", scale = 1)
+    private BigDecimal cvssV4BaseScore;
+
+    @Persistent
+    @Column(name = "CVSSV4THREATSCORE", scale = 1)
+    private BigDecimal cvssV4ThreatScore;
+
+    @Persistent
+    @Column(name = "CVSSV4ENVIRONMENTALSCORE", scale = 1)
+    private BigDecimal cvssV4EnvironmentalScore;
+
+    @Persistent
+    @Column(name = "CVSSV4VECTOR")
+    @JsonDeserialize(using = TrimmedStringDeserializer.class)
+    @Pattern(regexp = RegexSequence.Definition.PRINTABLE_CHARS_PLUS, message = "The CVSSv4 Vector may only contain printable characters")
```

Comment:
Maybe we should do here more strict validation, accept only valid CVSSv4 vectors

Reply:
It might make sense to look into creating a custom validator, like we have for [cron expressions](https://github.com/DependencyTrack/dependency-track/blob/07316d9b82e4c8e8fa68d2103540154665d2b820/src/main/java/org/dependencytrack/model/validation/CronExpressionValidator.java) for example.

I do agree that adding that for existing CVSS types might cause issues that would need proper data migration first. But for everything we add going forward, it makes sense to be more defensive as to what we accept.